jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -72,135 +67,41 @@ public abstract class Resource implements ResourceFactory
         .with("jar:")
         .build();
 
-    /**
-     * <p>Create a ResourceCollection from an unknown set of URIs.</p>
-     *
-     * <p>
-     *     Use this if you are working with URIs from an unknown source,
-     *     such as a user configuration.  As some of the entries
-     *     might need mounting, but we cannot determine that yet.
-     * </p>
-     *
-     * @param uris collection of URIs to mount into a {@code ResourceCollection}
-     * @return the {@link Mount} with a root pointing to the {@link ResourceCollection}
-     */
-    public static Resource.Mount mountCollection(Collection<URI> uris)
-    {
-        List<Resource> resources = new ArrayList<>();
-        List<Resource.Mount> mounts = new ArrayList<>();
-
-        try
-        {
-            // track URIs that have been seen, to avoid adding duplicates.
-            Set<URI> seenUris = new HashSet<>();
-
-            for (URI uri : uris)
-            {
-                if (seenUris.contains(uri))
-                    continue; // skip this one
-                Resource.Mount mount = Resource.mountIfNeeded(uri);
-                if (mount != null)
-                {
-                    mounts.add(mount);
-                    resources.add(mount.root()); // use mounted resource that has Path with proper FileSystem reference in it.
-                }
-                else
-                {
-                    resources.add(Resource.newResource(uri));
-                }
-                seenUris.add(uri);
-            }
-
-            return new ResourceCollection.Mount(resources, mounts);
-        }
-        catch (Throwable t)
-        {
-            // can't create ResourceCollection.Mount, so let's unmount and rethrow.
-            mounts.forEach(IO::close);
-            throw t;
-        }
-    }
-
     /**
      * <p>Mount a URI if it is needed.</p>
      *
      * @param uri The URI to mount that may require a FileSystem (e.g. "jar:file://tmp/some.jar!/directory/file.txt")
      * @return A reference counted {@link Mount} for that file system or null. Callers should call {@link Mount#close()} once
      * they no longer require any resources from a mounted resource.
      * @throws IllegalArgumentException If the uri could not be mounted.
-     * @see #mount(URI)
      */
-    public static Resource.Mount mountIfNeeded(URI uri)
+    static Resource.Mount mountIfNeeded(URI uri)
```

Comment:
Can this live in ResourceFactory?

Reply:
This went to `ResourceFactoryInternals`.